packages/message-manager/src/TypedMessageManager.ts

```diff
@@ -98,8 +98,9 @@ export class TypedMessageManager extends AbstractMessageManager<
       validateTypedSignMessageDataV1(messageParams);
     }
 
-    if (version === 'V3') {
-      validateTypedSignMessageDataV3(messageParams);
+    if (version === 'V3' || version === 'V4') {
```

Comment:
We should do the same validation for 'V4' messages.
https://github.com/MetaMask/metamask-extension/blob/033b776c0ed59d59a6ca7161b64daa5f76b2e461/app/scripts/lib/typed-message-manager.js#L199